tools/clang/tools/dxildll/dxccontainerbuilder.cpp

```diff
@@ -0,0 +1,24 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// dxccontainerbuilder.cpp                                                   //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Implements the Dxil Container Builder                                     //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#include "dxc/DxilContainer/DxcContainerBuilder.h"
+
+HRESULT CreateDxcHashingContainerBuilder(REFIID RRID, LPVOID *V) {
+  // Call dxil.dll's containerbuilder
+  *V = nullptr;
+  CComPtr<DxcContainerBuilder> Result(
+      DxcContainerBuilder::Alloc(DxcGetThreadMallocNoRef()));
+  if (nullptr == Result.p)
+    return E_OUTOFMEMORY;
+
+  Result->Init();
+  return Result->QueryInterface(RRID, V);
+}
```

Comment:
I'll pass along the [feedback ](https://github.com/microsoft/DirectXShaderCompiler/pull/5952#discussion_r1382231012)I got from #5952 and suggest that this file be moved into dxildll.cpp

Reply:
Updated.